core/src/apps/cardano/sign_tx/__init__.py

```diff
@@ -0,0 +1,35 @@
+from typing import Type
+
+from trezor import log, messages, wire
+from trezor.enums import CardanoTxSigningMode
+
+from .. import seed
+from .multisig_signer import MultisigSigner
+from .ordinary_signer import OrdinarySigner
+from .plutus_signer import PlutusSigner
+from .pool_owner_signer import PoolOwnerSigner
+from .signer import Signer
+
+
+@seed.with_keychain
+async def sign_tx(
+    ctx: wire.Context, msg: messages.CardanoSignTxInit, keychain: seed.Keychain
+) -> messages.CardanoSignTxFinished:
+    signer_types: dict[CardanoTxSigningMode, Type[Signer]] = {
```

Comment:
this _probably_ won't make any practical difference now, but it would be nicer to write this out like so:
```
if msg.signing_mode == CardanoSigningMode.ORDINARY_TRANSACTION:
    from .ordinary_signer import OrdinarySigner

    signer_type = OrdinarySigner
elif ...
```

so that the imports are conditional and we can save memory on parts that are not used for the given message type

Reply:
Fixed in https://github.com/trezor/trezor-firmware/pull/2313/commits/a1a246d283b61e12a724880aa215acac9d764d9e.